mne/annotations.py

```diff
@@ -430,28 +431,76 @@ def _write_annotations(fid, annotations):
     end_block(fid, FIFF.FIFFB_MNE_ANNOTATIONS)
 
 
-def read_annotations(fname):
-    """Read annotations from a FIF file.
+def read_annotations(fname, sfreq='auto', uint16_codec=None):
+    r"""Read annotations from a file.
+
+    This function reads a .fif, .fif.gz, .vrmk, .edf or .set file and makes an
+    :class:`mne.Annotations` object.
 
     Parameters
     ----------
     fname : str
         The filename.
 
+    sfreq : float | 'auto'
+        The sampling frequency in the file. This parameter is necessary for
+        \*.vmrk files as Annotations are expressed in seconds and \*.vmrk files
+        are in samples. For any other file format, ``sfreq`` is omitted.
+        If set to 'auto' then the ``sfreq`` is taken from the \*.vhdr
+        file that has the same name (without file extension). So data.vrmk
+        looks for sfreq in data.vhdr.
+
+    uint16_codec : str | None
+        This parameter is only used in EEGLAB (\*.set) and omitted otherwise.
+        If your \*.set file contains non-ascii characters, sometimes reading
+        it may fail and give rise to error message stating that "buffer is
+        too small". ``uint16_codec`` allows to specify what codec (for example:
+        'latin1' or 'utf-8') should be used when reading character arrays and
+        can therefore help you solve this problem.
+
     Returns
     -------
     annot : instance of Annotations | None
         The annotations.
     """
-    ff, tree, _ = fiff_open(fname, preload=False)
-    with ff as fid:
-        annotations = _read_annotations(fid, tree)
+    # XXX: I've issues with circular imports
```

Comment:
No need for XXX here, we should nest our own imports if and only there is a circular problem

But these should be relative not absolute imports, e.g. `from .io.brainvision...`